src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -200,76 +188,73 @@ public boolean isCommentNodesRequired() {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final int lineNo = ast.getLineNo();
-        if (illegalLines.get(lineNo) != null
-                && illegalLines.get(lineNo).equals(ast.getColumnNo())) {
-            log(ast, MSG_KEY);
-        }
-    }
+        final String commentContent = ast.getFirstChild().getText();
+        final boolean isCommentLegal;
 
-    @Override
-    public void beginTree(DetailAST rootAST) {
-        final Map<Integer, TextBlock> cppComments = getFileContents()
-                .getSingleLineComments();
-        final Map<Integer, List<TextBlock>> cComments = getFileContents()
-                .getBlockComments();
-        final Set<Integer> lines = new HashSet<>();
-        lines.addAll(cppComments.keySet());
-        lines.addAll(cComments.keySet());
+        if (ast.getType() == TokenTypes.SINGLE_LINE_COMMENT) {
+            isCommentLegal = isCommentMatchLegalPattern(commentContent);
```

Comment:
there is no need to do it like this. It is still duplication (less than before, but it is)
I suggest following:
```
final boolean isCommentLegal = isCommentMatchLegalPattern(commentContent);
final boolean isSingleLine = ast.getType() == TokenTypes.SINGLE_LINE_COMMENT 
|| isSingleLineBlockComment(); <- all logic from else block goes here
if (!isCommentLegal && !isBeforeCommentLegal(ast) && isSingleLine) {
   log();
}
```

Reply:
fixed. Great suggestion!

Determining a comment is `legal` is different for `singleLine` and `blockComment`

singleLineComment is legal if the `content` matches `legalPattern`

blockComment is legal if one of the following is true:
- The comment is `not end of line`. (Eg: `Thread.sleep( 10 /*some comment here*/ );`)
- The comment is `single line` and it matches `legalPattern`

I merged these two logic into one to avoid duplication. I have one minor concern though: I get `commentEnd` twice, one in `isBlockCommentEndOfLine` and one in `isBlockCommentOnOneLine`. I cannot move `commentEnd` to the `visitToken()` since only `blockComment` has `commentEnd`. I'm wondering if this is ok?